### ISSUE TITLE
"Open New York Times bookmark" opens MDN if the first article from nytimes.com is bookmarked

### ISSUE THREAD

**Issue by alexandra-martin** [mozilla/firefox-voice#456]
### Prerequisites:
Mic permissions are enabled.
Have the first New York Times [article](https://www.nytimes.com/2019/10/22/us/politics/trump-ukraine-orban.html?action=click&module=Top%20Stories&pgtype=Homepage) from current day bookmarked.

### STR:
1. Use the shortcut or click on the mic icon from the browser toolbar.
2. Say or write an "Open bookmark" command, e.g. "Open New York Times bookmark".

### Expected result:
The bookmarked New York Times article opens in a new tab.

### Actual result:
MDN opens in a new tab.

### Notes:
Reproduced on Mac 10.14.6 and Win10x64 with Firefox Nightly 72.0a1 (64-bit).
Using commands like "Open nytimes bookmark" or "Open Hungary bookmark" opens the bookmarked article in a new tab. (2nd gif)
Other bookmarked articles open with the "Open New York Times bookmark" command. (3rd gif)

![6](https://user-images.githubusercontent.com/54401091/67282763-1e9afe00-f4db-11e9-86aa-57e4eeb19fa0.gif)

![7](https://user-images.githubusercontent.com/54401091/67282781-25297580-f4db-11e9-9c10-280b754a0c92.gif)

![8](https://user-images.githubusercontent.com/54401091/67282798-29559300-f4db-11e9-91bf-6204a9a43630.gif)

**Comment by awallin** [comment]
@alexandra-martin Do you have an MDN bookmark?

**Comment by alexandra-martin** [comment]
@awallin yeah, it seems there are some standard Mozilla related bookmarks saved in Firefox Nightly, MDN being one of them. Once I deleted that bookmark, the New York Times bookmark mentioned in prerequisites opened with the "Open New York Times bookmark" command. But I'm curious as to why MDN was opening only for that article with that command and it wasn't opening for other bookmark articles.

**Comment by ianb** [comment]
I'm not sure how easy this is to reproduce, but if you can then this is probably a bug in `bookmarks.js`, maybe with how the Fuse search is handled.

**Comment by Simpcyclassy** [comment]
Hello, I am an outreachy applicant. May I go ahead and work on this bug?

**Comment by ianb** [comment]
@Simpcyclassy You got it! I'd recommend focusing on reproducing the issue first.

**Comment by Simpcyclassy** [comment]
OK. Thank you. will reach out once I do

**Comment by Simpcyclassy** [comment]
![ezgif com-video-to-gif](https://user-images.githubusercontent.com/34488930/76167454-0f091280-6167-11ea-9b28-4de1224aa68a.gif)


case 1

**Comment by Simpcyclassy** [comment]
![ezgif com-video-to-gif (1)](https://user-images.githubusercontent.com/34488930/76167697-f1d54380-6168-11ea-827f-f0c73d03939b.gif)


case 2

**Comment by Simpcyclassy** [comment]
![ezgif com-video-to-gif (2)](https://user-images.githubusercontent.com/34488930/76167800-c1da7000-6169-11ea-9cc2-cbcbc80b8ed7.gif)


Case 3, only the second instance opened the new York times, other times open MDN

**Comment by Simpcyclassy** [comment]
@ianb Please, could you take a look at this?

**Comment by ianb** [comment]
@Simpcyclassy Case 1 looks like you are reproducing the bug, so that looks like a good place to focus. 

`extension/intents/bookmarks/bookmarks.js` is where this is implemented, and you'll want to look at this code to try to understand what bookmarks it is searching and why it selects the wrong one: https://github.com/mozilla/firefox-voice/blob/master/extension/intents/bookmarks/bookmarks.js#L12-L43

**Comment by Simpcyclassy** [comment]
@ianb So I think it had something to do with the particular bookmark firefox nightly came with. I deleted the bookmark and readded and the bug is gone. The bookmark works as should

**Comment by ianb** [comment]
Cool. @Simpcyclassy if you need to get back to a fresh profile, deleting the `Profile/` directory will reset everything.

**Comment by Simpcyclassy** [comment]
Thank you! Here is my PR https://github.com/mozilla/firefox-voice/pull/1209.

**Comment by alexandra-martin** [comment]
Verified on Mac 10.14 and Win10x64 with Firefox Nightly 76.0a1 (2020-03-13).
When I first started to verify, I was able to reproduce the issue on Mac. After making several attempts and after opening several new Firefox profiles I can't reproduce the issue anymore. Will update, if I encounter the bug again. 

![3](https://user-images.githubusercontent.com/54401091/76624466-0a18da00-653e-11ea-90e5-a3b9cae9fc03.gif)
